api/utils/conv.go

```diff
@@ -20,10 +20,88 @@ package utils
 import (
 	"bytes"
 	"encoding/json"
+	"math"
+	"strconv"
 
 	"github.com/gravitational/trace"
 )
 
+// IntToUInt16 converts an int to uint16, checking for possible overflow or underflow issues.
+func IntToUInt16(i int) (uint16, error) {
```

Comment:
I'm not a big fan of general purpose functions becoming part of Teleport's public API module. If this is necessary (and I'm not totally convinced of that yet, either), it seems like more of an implementation detail than functions we'd want to expose to people building things with Teleport.

tl;dr, it's hard to remove things from a public API once they exist, so best to keep the surface area as small as reasonably possible.